src/PureScript/CST/Traversal.purs

```diff
@@ -221,14 +222,20 @@ traverseInstance
    . Applicative f
   => { | OnBinder (Rewrite e f) + OnExpr (Rewrite e f) + OnType (Rewrite e f) + r }
   -> Rewrite e f Instance
-traverseInstance k = Newtype.traverse Instance (\i -> i { head = _, body = _ } <$> traverseInstanceHead k i.head <*> traverse (traverse (traverse (traverseInstanceBinding k))) i.body)
+traverseInstance k (Instance i) =
+  map Instance $ i { head = _, body = _ }
```

Comment:
These should be something like `\head body -> Instance i { head = head, body = body }` otherwise you are double mapping unnecessarily.

Reply:
Good point; I'll update it.